cluster/kubernetes/cached_disco.go

```diff
@@ -0,0 +1,104 @@
+package kubernetes
+
+import (
+	"sync"
+	"time"
+
+	crdv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	crd "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/discovery"
+	discocache "k8s.io/client-go/discovery/cached"
+	toolscache "k8s.io/client-go/tools/cache"
+)
+
+// This mainly exists so I can put the controller and store in here,
+// so as to have access to them for testing; and, so that we can do
```

Comment:
The first part of the comment doesn't apply anymore

Reply:
Oh yes. oops